daft/execution/rust_physical_plan_shim.py

```diff
@@ -314,6 +315,12 @@ def broadcast_join(
     )
 
 
+def _partition_cols_to_projection(partition_cols: list[PyExpr] | None) -> ExpressionsProjection | None:
+    if partition_cols is not None:
+        return ExpressionsProjection([Expression._from_pyexpr(expr) for expr in partition_cols])
+    return None
```

Comment:
Why pull this out? Seems it's only used in 1 place

Reply:
Gonna use it in the delta partitioned writing PR. I'll just remove it for now